### PR TITLE
Feature: Add material setting nozzle size and add material color selector

### PR DESCRIPTION
**Pull request by bbaa77770**
[LU-1546](https://snapmaker2.atlassian.net/browse/LU-1546) part2
[LU-1892](https://snapmaker2.atlassian.net/browse/LU-1892)

### CODE REVIEW

File: src/app/flux/printing/index.js [hunk 1]
```diff
@@ -1597,7 +1597,8 @@ export const actions = {
                         material: material,
                         transformation,
                         modelID,
-                        extruderConfig
+                        extruderConfig,
+                        color: activeDefinition.settings.color.default_value
```
Comment (jane-rose) [inline src/app/flux/printing/index.js:1601]: default color

**Review by jane-rose — APPROVED** [review]

File: src/app/flux/printing/index.js [hunk 2]
```diff
@@ -960,11 +1001,30 @@ export const actions = {
...
+        }));
+        definitionManager.updateDefinition({
+            ...newExtruderLDefinition,
+            definitionId: 'snapmaker_extruder_0'
+        });
+        definitionManager.updateDefinition({
+            ...newExtruderRDefinition,
+            definitionId: 'snapmaker_extruder_1'
+        });
+
```
Comment (jane-rose) [inline src/app/flux/printing/index.js:1027]: Distinguish between single head and double head？

File: src/server/slicer/slice.js [hunk 3]
```diff
@@ -46,6 +46,7 @@ function callCuraEngine(modelConfig, supportConfig, outputPath) {
             args.push('-j', supportConfig.configFilePath);
         }
     }
+    console.log(args.join(' '));
```
Comment (jane-rose) [inline src/server/slicer/slice.js:49]: log

File: src/app/ui/views/ProfileManager/SettingItem.jsx [hunk 4]
```diff
@@ -191,8 +197,37 @@ function SettingItem({ definitionKey, settings, isDefaultDefinition = () => true
                     {type === undefined && (
                         <span className="sm-parameter-row__input-unit">{unit}</span>
                     )}
+                    {type === 'color' && (
+                        <div
+                            className="sm-flex-width align-r height-percent-100 width-96 display-inline border-radius-8"
+                            role="button"
+                            tabIndex="-1"
+                            onKeyPress={() => {}}
+                            style={{
```
Comment (xiekun1992) [inline src/app/ui/views/ProfileManager/SettingItem.jsx:206]: use `class`

File: src/app/ui/widgets/PrintingMaterial/MaterialWithColor.jsx [hunk 5]
```diff
@@ -5,7 +5,14 @@ export const MaterialWithColor = ({ name, color }) => {
     return (
         <div className="sm-flex align-center justify-space-between">
             <span>{name}</span>
-            <div className={`width-16 height-16 material-background-${color?.toLowerCase()}`} />
+            <div
+                className="width-16 height-16"
+                style={{
```
Comment (xiekun1992) [inline src/app/ui/widgets/PrintingMaterial/MaterialWithColor.jsx:10]: use `class`

File: src/app/models/ThreeModel.js [hunk 6]
```diff
@@ -96,6 +85,8 @@ class ThreeModel extends BaseModel {
                 this.transformation.positionY = point.y;
             }
         }
+
+        this.updateMaterialColor(modelInfo.color ?? '#cecece');
```
Comment (xiekun1992) [inline src/app/models/ThreeModel.js:89]: use constant?

**Review by xiekun1992 — CHANGES_REQUESTED** [review]

File: src/app/ui/views/ProfileManager/index.jsx [hunk 7]
```diff
@@ -67,6 +68,7 @@ function useGetDefinitions(allDefinitions, definitionState, setDefinitionState,
             checkboxAndSelectGroup.label = d.name;
             checkboxAndSelectGroup.value = d.definitionId;
             checkboxAndSelectGroup.isDefault = !!d.isDefault;
+            checkboxAndSelectGroup.color = d?.settings?.color?.default_value;
```
Comment (jane-rose) [inline src/app/ui/views/ProfileManager/index.jsx:71]: default_value could't use `?.`

File: src/app/flux/manager/DefinitionManager.js [hunk 8]
```diff
@@ -200,7 +215,7 @@ class DefinitionManager {
         }
         if (settings.support_infill_rate) {
             const supportInfillRate = settings.support_infill_rate.default_value;
-            const supportLineWidth = definition.settings.support_line_width.default_value;
+            const supportLineWidth = extruderLDefinitionSettings.machine_nozzle_size.default_value; // support_line_width
```
Comment (jane-rose) [inline src/app/flux/manager/DefinitionManager.js:217]: just using `extruderLDefinitionSettings`?

File: src/app/flux/manager/DefinitionManager.js [hunk 9]
```diff
@@ -239,9 +254,17 @@ class DefinitionManager {
...
+            settings,
+            extruderLDefinitionSettings,
+            extruderRDefinitionSettings
+        };
     }
 
+    // calculateExtruderDependencies(definition, settings) {
+    //
+    // }
+
```
Comment (jane-rose) [inline src/app/flux/manager/DefinitionManager.js:266]: remove it

File: src/server/slicer/slice.js (hunk 3 above)
Comment (parachvte) [reply src/server/slicer/slice.js:49]: No resolved but PR merged?

**Pull request merged**